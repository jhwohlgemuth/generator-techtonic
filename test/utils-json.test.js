'use strict';

const {join}       = require('path');
const {unlinkSync} = require('fs');
const {json}       = require('../generators/app/utils');

describe('Generator JSON utility', () => {
    let content;
    let testJSON = join(__dirname, 'test.json');
    beforeEach(() => {
        json.write(testJSON, {foo: 'bar'});
    });
    afterEach(() => {
        unlinkSync(testJSON);
    });
    it('can extend JSON files', () => {
        content = json.read(testJSON);
        expect(content).toMatchSnapshot();
        json.extend(testJSON, {bar: 'baz'});
        content = json.read(testJSON);
        expect(content).toMatchSnapshot();
    });
});
