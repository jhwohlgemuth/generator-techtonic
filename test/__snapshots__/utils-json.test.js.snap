// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Generator JSON utility can extend JSON files 1`] = `
Object {
  "foo": "bar",
}
`;

exports[`Generator JSON utility can extend JSON files 2`] = `
Object {
  "bar": "baz",
  "foo": "bar",
}
`;
